rest-ehr-scape/src/main/java/org/ehrbase/rest/ehrscape/controller/EhrController.java

```diff
@@ -156,30 +157,38 @@ private EhrStatus extractEhrStatus(@RequestBody String content) {
         ehrStatus.setArchetypeNodeId("openEHR-EHR-EHR_STATUS.generic.v1");
         ehrStatus.setName(new DvText("EHR Status"));
 
-        if (StringUtils.isNotBlank(content)) {
-            Gson json = new GsonBuilder().create();
-            Map<String, Object> atributes = json.fromJson(content, Map.class);
-
-            Optional<String> subjectId = Optional.ofNullable(atributes.get("subjectId"))
-                    .map(String.class::cast)
-                    .filter(StringUtils::isNotBlank);
-            Optional<String> subjectNamespace = Optional.ofNullable(atributes.get("subjectNamespace"))
-                    .map(String.class::cast)
-                    .filter(StringUtils::isNotBlank);
-            if (subjectId.isEmpty() || subjectNamespace.isEmpty()) {
-                throw new InvalidApiParameterException("subjectId or subjectNamespace missing");
-            }
-            PartySelf subject =
-                    new PartySelf(new PartyRef(new HierObjectId(subjectId.get()), subjectNamespace.get(), "PERSON"));
-            ehrStatus.setSubject(subject);
+        if (StringUtils.isBlank(content)) {
+            return ehrStatus;
+        }
 
-            if (atributes.containsKey(MODIFIABLE)) {
-                ehrStatus.setModifiable((Boolean) atributes.get(MODIFIABLE));
-            }
+        Map<String, Object> attributes;
+        try {
+            attributes = new ObjectMapper().readerForMapOf(Object.class).readValue(content);
+        } catch (JsonProcessingException e) {
+            throw new GeneralRequestProcessingException("Invalid content format", e);
+        }
 
-            if (atributes.containsKey(QUERYABLE)) {
-                ehrStatus.setQueryable((Boolean) atributes.get(QUERYABLE));
-            }
+        String subjectId = Optional.of("subjectId")
```

Comment:
Please check Optional.of("subjectId") part

Reply:
Do you mean if the literal is null?